pkg/scheduler/framework/v1alpha1/framework_test.go

```diff
@@ -20,160 +20,105 @@ import (
 	"reflect"
 	"testing"
 
-	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/kubernetes/pkg/scheduler/apis/config"
 )
 
 const (
+	scoreWithNormalizePlugin1  = "score-with-normalize-plugin-1"
+	scoreWithNormalizePlugin2  = "score-with-normalize-plugin-2"
 	scorePlugin1               = "score-plugin-1"
-	scorePlugin2               = "score-plugin-2"
-	scorePlugin3               = "score-plugin-3"
 	pluginNotImplementingScore = "plugin-not-implementing-score"
-	weight1                    = 2
-	weight2                    = 3
-	weight3                    = 4
 )
 
-// TestScorePlugin1 and 2 implements ScoreWithNormalizePlugin interface,
-// TestScorePlugin3 only implements ScorePlugin interface.
-var _ = ScoreWithNormalizePlugin(&TestScorePlugin1{})
-var _ = ScoreWithNormalizePlugin(&TestScorePlugin2{})
-var _ = ScorePlugin(&TestScorePlugin3{})
+// ScoreWithNormalizePlugin1 and 2 implements ScoreWithNormalizePlugin interface,
+// ScorePlugin1 only implements ScorePlugin interface.
+var _ = ScoreWithNormalizePlugin(&ScoreWithNormalizePlugin1{})
+var _ = ScoreWithNormalizePlugin(&ScoreWithNormalizePlugin2{})
+var _ = ScorePlugin(&ScorePlugin1{})
 
-type TestScorePlugin1 struct {
-	// If fail is true, NormalizeScore will return error status.
-	fail bool
-}
-
-// NewScorePlugin1 is the factory for NormalizeScore plugin 1.
-func NewScorePlugin1(_ *runtime.Unknown, _ FrameworkHandle) (Plugin, error) {
-	return &TestScorePlugin1{}, nil
-}
+func NewScoreWithNormalizePlugin1(inj injectedResult) Plugin { return &ScoreWithNormalizePlugin1{inj} }
+func NewScoreWithNormalizePlugin2(inj injectedResult) Plugin { return &ScoreWithNormalizePlugin2{inj} }
+func NewScorePlugin1(inj injectedResult) Plugin              { return &ScorePlugin1{inj} }
+func NewPluginNotImplementingScore(injectedResult) Plugin    { return &PluginNotImplementingScore{} }
 
-// NewScorePlugin1InjectFailure creates a new TestScorePlugin1 which will
-// return an error status for NormalizeScore.
-func NewScorePlugin1InjectFailure(_ *runtime.Unknown, _ FrameworkHandle) (Plugin, error) {
-	return &TestScorePlugin1{fail: true}, nil
+type ScoreWithNormalizePlugin1 struct {
```

Comment:
It seems that `ScoreWithNormalizePlugin1` and `ScoreWithNormalizePlugin2` are almost same except name, could we create a generic `ScoreWithNormalizePlugin` for them?

Reply:
Yeah they are pretty much the same. The intention was just to test multiple plugin case, no other reason beyond that. What do you think?

Reply:
e.g.

```
type ScoreWithNormalizePlugin struct {
        name string
        inj injectedResult
}

func (pl *ScoreWithNormalizePlugin) Name() string {
	return pl.name
}

func (pl *ScoreWithNormalizePlugin) NormalizeScore(pc *PluginContext, pod *v1.Pod, scores NodeScoreList) *Status {
	return injectNormalizeRes(pl.inj, scores)
}

func (pl *ScoreWithNormalizePlugin) Score(pc *PluginContext, p *v1.Pod, nodeName string) (int, *Status) {
	return setScoreRes(pl.inj)
}
```
Then we could create `ScoreWithNormalizePlugin1` with `&ScoreWithNormalizePlugin1{"score-with-normalize-plugin-1", inj}` and `ScoreWithNormalizePlugin1` with `&ScoreWithNormalizePlugin1{"score-with-normalize-plugin-2", inj}`.

Reply:
Oh sorry I misunderstood your question. Done.